code/modules/projectiles/guns/projectile/revolver.dm

```diff
@@ -114,4 +114,85 @@
 	recoil = 2 // ow my fucking hand
 	accuracy = -1 // smooth bore + short barrel = shit accuracy
 	ammo_type = /obj/item/ammo_casing/shotgun
-	// ToDo: Remove accuracy debuf in exchange for slightly injuring your hand every time you fire it.
\ No newline at end of file
+	// ToDo: Remove accuracy debuf in exchange for slightly injuring your hand every time you fire it.
+
+/obj/item/weapon/gun/projectile/revolver/lemat
+	name = "LeMat Revolver"
+	desc = "The LeMat Revolver is a 9 shot revolver with a secondary firing barrel loading shotgun shells. For when you really need something dead."
+	icon_state = "lemat"
+	item_state = "revolver"
+	origin_tech = list(TECH_COMBAT = 2, TECH_MATERIAL = 2)
+	handle_casings = CYCLE_CASINGS
+	max_shells = 9
+	caliber = "38"
+	ammo_type = /obj/item/ammo_casing/c38
+	var/secondary_max_shells = 1
+	var/secondary_caliber = "shotgun"
+	var/secondary_ammo_type = /obj/item/ammo_casing/shotgun
+	var/flipped_firing = 0
+	var/list/secondary_loaded = list()
+	var/list/tertiary_loaded = list()
+
+
+/obj/item/weapon/gun/projectile/revolver/lemat/New()
+	world << "1"
+	for(var/i in 1 to secondary_max_shells)
+		world << "2"
+		secondary_loaded += new secondary_ammo_type(src)
+	..()
+
+/obj/item/weapon/gun/projectile/revolver/lemat/verb/swap_firingmode()

```

Comment:
I could have. But Revolvers have attack_hand to be unload all the ammo in the gun, while clicking it with your other hand removes one casing. I kinda wanted to retain that because it's somewhat necessary for unloading the gun when you're done shooting. Either way, I'd have to either have my own thing, like here, or override one of the attack things to make it unload all the ammo at once when clicked with other hand rather than one.

The Bulldog is probably quite a bit better done, but I've still got a few things to add. Like swapping firemodes automatically when shooting the shotgun underbarrel.